include/vcpkg/base/stringview.h

```diff
@@ -86,3 +88,45 @@ namespace vcpkg
 
 VCPKG_FORMAT_AS(vcpkg::ZStringView, vcpkg::StringView);
 VCPKG_FORMAT_AS(vcpkg::StringLiteral, vcpkg::StringView);
+
+template<::size_t N>
+struct StringArray : public std::array<char, N>
+{
+    constexpr StringArray() noexcept : std::array<char, N>() { }
+    constexpr StringArray(const char (&str)[N]) noexcept : std::array<char, N>()
+    {
+        for (size_t i = 0; i < N; i++)
+        {
+            this->at(i) = str[i];
+        }
+    }
+
+    constexpr ::size_t length() const noexcept { return this->size() - 1; }
+
+    template<::size_t L, ::size_t R>
+    friend constexpr StringArray<L + R - 1> operator+(const StringArray<L> lhs, const StringArray<R> rhs) noexcept;
+};
+
+template<::size_t L, ::size_t R>
+constexpr StringArray<L + R - 1> operator+(const StringArray<L> lhs, const StringArray<R> rhs) noexcept
+{
+    if constexpr (lhs.length() == 0)
+        return rhs;
+    else if constexpr (rhs.length() == 0)
+        return lhs;
+    else
+    {
+        StringArray<L + R - 1> out;
+
+        for (size_t i = 0; i < lhs.length(); i++)
+        {
+            out[i] = lhs[i];
```

Comment:
Can't use `std::copy` because it's not `constexpr` in C++17

Reply:
Add:

```
template<class InIt, class OutIt>
constexpr OutIt constexpr_copy(InIt first, InIt last, OutIt dest) {
  while (first != last) {
    *dest = *first;
    ++dest;
    ++first;
  }

  return dest;
}
```

and use that?